src/initRules.js

```diff
@@ -138,15 +126,23 @@ module.exports = function(
           var timeoutId;
 
           return new Promise(function(resolve, reject) {
+            var promiseTimeout = condition.settings.timeout;
+
+            if (isNaN(promiseTimeout) || promiseTimeout < 0) {
+              reject(new Error('Condition timeout is not correctly defined.'));
+            }
```

Comment:
Will this rejection occur if a customer creates a condition without a timeout? It does seem odd that we would require the customer to define a timeout, particularly in the case where they know it's a synchronous condition. Maybe it would be better to assume a large default timeout instead, like 10 seconds?

Reply:
Since Lens will always emit timeout for conditions, I think I prefer to leave it like this. I was thinking about this error only from the sandbox and rule editor perspective (it will help catch errors more easily).

Reply:
We would be requiring API users to provide a timeout even if the condition is synchronous then? This seems wrong.

Reply:
Is there something in the Lens where they will be able to define if a condition is sync or async?

Reply:
No. I'm saying we're allowing the customer to get themselves into problematic situations where we could otherwise be helpful. If I understand correctly, we're not planning on throwing an error when the customer tries to create the condition without a timeout using the API, but they _will_ get an error when they run that condition in Turbine. That's probably going to be an unexpected behavior. My point was, there's an even greater likelihood that it's an unexpected behavior if the condition the customer is created is something that they could reasonably consider would be synchronous, like whether the URL matches XYZ. Rather than throwing an error if they don't provide a timeout, could it be reasonable to assume a timeout on their behalf (perhaps use whatever Lens will use)? I think so.

What were you planning to do for existing conditions that have been created in Launch? Were you planning to migrate them all to have a `timeout` defined?

Reply:
Yes, all the conditions and actions will have a timeout defined.

I've removed the code and the tests related timeout.

Reply:
I'm confused. Why are they optional in the container schema then? And I thought actions would only have a `timeout` defined if the customer wanted to delay subsequent actions...